From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 22 Sep 2020 11:05:12 +0200
Subject: ath11k: Disable intersection of regd from BDF/firmware

We definitely don't want to use an arbitrary default domain which is
intersected with each other. Instead, the each reported regulary domain
from the device should be handled independently.

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/drivers/net/wireless/ath/ath11k/wmi.c b/drivers/net/wireless/ath/ath11k/wmi.c
index 54617c0e5484700939417abaa6652524dda29c36..ad31e709fc78978376178a38ef4ca51d6b6c8ebe 100644
--- a/drivers/net/wireless/ath/ath11k/wmi.c
+++ b/drivers/net/wireless/ath/ath11k/wmi.c
@@ -6786,16 +6786,6 @@ static int ath11k_reg_chan_list_event(struct ath11k_base *ab, struct sk_buff *sk
 		    (char *)reg_info->alpha2, 2))
 		goto mem_free;
 
-	/* Intersect new rules with default regd if a new country setting was
-	 * requested, i.e a default regd was already set during initialization
-	 * and the regd coming from this event has a valid country info.
-	 */
-	if (ab->default_regd[pdev_idx] &&
-	    !ath11k_reg_is_world_alpha((char *)
-		ab->default_regd[pdev_idx]->alpha2) &&
-	    !ath11k_reg_is_world_alpha((char *)reg_info->alpha2))
-		intersect = true;
-
 	regd = ath11k_reg_build_regd(ab, reg_info, intersect);
 	if (!regd) {
 		ath11k_warn(ab, "failed to build regd from reg_info\n");
